File: c/test/test_large_numbers.c
#include "test_utils.h"
#include <emtrace/emtrace.h>
#include <limits.h>
#include <stdint.h>

// Note: C doesn't have native 128-bit integers like Rust,
// so we'll use 64-bit integers as the largest available type
const char g_emt_expected_output[] EMT_TEST_EXPECTED_SECTION =
    "Large integers:\n"
    "int64_t max: 9223372036854775807\n"
    "uint64_t max: 18446744073709551615\n"
    "int64_t min: -9223372036854775808\n";

int main(void) {
    EMTRACE_INIT();

    EMTRACELN("Large integers:");

    // Test maximum 64-bit signed integer
    int64_t max_i64 = INT64_MAX; // 9223372036854775807
    EMTRACELN_F("int64_t max: {}", int64_t, max_i64);

    // Test maximum 64-bit unsigned integer
    uint64_t max_u64 = UINT64_MAX; // 18446744073709551615
    EMTRACELN_F("uint64_t max: {}", uint64_t, max_u64);

    // Test minimum 64-bit signed integer
    int64_t min_i64 = INT64_MIN; // -9223372036854775808
    EMTRACELN_F("int64_t min: {}", int64_t, min_i64);

    return 0;
}

File: rust/examples/test_large_numbers.rs
use emtrace::{Out, expect, init, traceln};

fn main() {
    expect!(
        b"Large integers:\n\
          i128: 170141183460469231731687303715884105727\n\
          u128: 340282366920938463463374607431768211455\n\
          Negative i128: -170141183460469231731687303715884105728\n"
    );

    init(&mut std::io::stdout().lock());

    traceln!("Large integers:");

    // Test 128-bit integers
    traceln!("i128: {}", i128: i128::MAX);
    traceln!("u128: {}", u128: u128::MAX);
    traceln!("Negative i128: {}", i128: i128::MIN);
}
